amp/WEB-INF/src/org/digijava/module/aim/action/CompareActivityVersions.java

```diff
@@ -352,10 +353,38 @@ private void setAdvancemode(CompareActivityVersionsForm vForm, HttpServletReques
     public ActionForward viewDifferences(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
 
         CompareActivityVersionsForm vForm = (CompareActivityVersionsForm) form;
-        vForm.setOutputCollection(new ArrayList<CompareOutput>());
         vForm.setOutputCollectionGrouped(ActivityVersionUtil.compareActivities(vForm.getActivityOneId()));
 
         return mapping.findForward("forward");
     }
 
+    public ActionForward compareAll(ActionMapping mapping, ActionForm form, HttpServletRequest request,
+                                    HttpServletResponse response) throws Exception {
+
+        CompareActivityVersionsForm vForm = (CompareActivityVersionsForm) form;
+
+        // We may have duplicate keys.. so time to switch to something like Multimap
```

Comment:
If we have duplicate keys, we need to solve that at query level to avoid getting more results than needed. if we use a multi map,then we add overhead to the program (its slow and needs to be optimized, but we add a burden to remove duplicate objects). If the criteria returns dupes, please work it out to avoid them